### ISSUE TITLE
Syntax highlighting bugs

### ISSUE THREAD

**Issue by mhuisi** [leanprover-community/vscode-lean4#10]
Opening this issue to document some syntax highlighting bugs. Issues found so far:

1. `deriving instance <...> for <...>` or the empty inductive `inductive <...>` tags the rest of the file as a declaration name
2. Code blocks within comments tag the rest of the file as a comment

**Comment by gebner** [comment]
> Code blocks within comments tag the rest of the file as a comment

Do you mean unclosed code blocks?
````lean
/--
```
-/
def foo := 42
````

This is extremely hard (maybe even impossible) to get right using the syntax highlighter in vscode.

**Comment by gebner** [comment]
Yes, as I've written above there's very little we can realistically do about this.
